ptf/sainat.py

```diff
@@ -0,0 +1,1493 @@
+            destination=sai_ipprefix(self.dst_ip + mask))
+        sai_thrift_create_route_entry(client=self.client,
+                                      route_entry=route_entry1,
+                                      next_hop_id=nhop1)
+        self.route_list.append(route_entry1)
+
+        route_entry2 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(self.nhop_ip + mask))
+        sai_thrift_create_route_entry(client=self.client,
+                                      route_entry=route_entry2,
+                                      next_hop_id=wan_rif)
+        self.route_list.append(route_entry2)
+
+        # 192.168.0.1/24 --> NHOP(SERVER)
+        print("Create nieghbor with %s ip address, %d router interface id and "
+              "%s destination mac"
+              % (self.server_ip, server_rif, self.server_dmac))
+        nbr_entry2 = sai_thrift_neighbor_entry_t(
+            rif_id=server_rif,
+            ip_address=sai_ipaddress(self.server_ip))
+        sai_thrift_create_neighbor_entry(client=self.client,
+                                         neighbor_entry=nbr_entry2,
+                                         dst_mac_address=self.server_dmac)
+        self.nbor_list.append(nbr_entry2)
+
+        print("Create nhop with %s ip address and %d router interface id"
+              % (self.server_ip, server_rif))
+        nhop2 = sai_thrift_create_next_hop(
+            self.client, ip=sai_ipaddress(self.server_ip),
+            router_interface_id=server_rif,
+            type=SAI_NEXT_HOP_TYPE_IP)
+        self.nhop_list.append(nhop2)
+
+        route_entry3 = sai_thrift_route_entry_t(
+            vr_id=self.default_vrf,
+            destination=sai_ipprefix(self.server_ip + mask))
+        sai_thrift_create_route_entry(client=self.client,
+                                      route_entry=route_entry3,
+                                      next_hop_id=nhop2)
+        self.route_list.append(route_entry3)
+
+        # NAT configuration
+        # Translated server IP - 20.20.20.1
+        # WAN IP - 10.10.10.1
+        # self.server_ip = '192.168.0.1'
+
+        l4_src_port = 100
+        l4_dst_port = 1234
+        proto = 6
+
+        translate_sport = 500
+        translate_dport = 2000
+
+        nat_type = SAI_NAT_TYPE_DESTINATION_NAT
+        nat_data = sai_thrift_nat_entry_data_t(
+            key=sai_thrift_nat_entry_key_t(
+                dst_ip=self.translate_server_ip,
+                proto=proto,
+                l4_dst_port=l4_dst_port))
+        dnat1 = sai_thrift_nat_entry_t(vr_id=self.default_vrf,
+                                       data=nat_data,
+                                       nat_type=nat_type)
+        sai_thrift_create_nat_entry(self.client,
+                                    nat_entry=dnat1,
+                                    nat_type=nat_type,
+                                    dst_ip=self.server_ip,
```

Comment:
Do you want to add a comment here to reflect this is same as sai attribute: SAI_NAT_ENTRY_ATTR_DST_IP and SAI_NAT_ENTRY_ATTR_L4_DST_PORT.